### PR TITLE
Extend RabbitMQ instance parameters

### PR DESCRIPTION
**Pull request by DiogoFerrao**

### CODE REVIEW

File: stackit/internal/services/rabbitmq/instance/resource.go [hunk 1]
```diff
@@ -124,6 +148,22 @@ func (r *instanceResource) Schema(_ context.Context, _ resource.SchemaRequest, r
...
+		"graphite":               "Graphite server URL (host and port). If set, monitoring with Graphite will be enabled.",
+		"max_disk_threshold":     "The maximum disk threshold in MB. If the disk usage exceeds this threshold, the instance will be stopped.",
+		"metrics_frequency":      "The frequency in seconds at which metrics are emitted.",
+		"metrics_prefix":         "The prefix for the metrics. Could be useful when using Graphite monitoring to prefix the metrics with a certain value, like an API key",
+		"monitoring_instance_id": "The monitoring instance ID.",
+		"plugins":                "List of plugins to install. Must be a supported plugin name.",
+		"roles":                  "List of roles to assign to the instance.",
+		"syslog":                 "List of syslog servers to send logs to.",
+		"tls_ciphers":            "List of TLS ciphers to use.",
+		"tls_protocols":          "TLS protocol to use.",
```
Comment (joaopalet) [inline stackit/internal/services/rabbitmq/instance/resource.go:164]: 👍

File: stackit/internal/services/rabbitmq/instance/resource.go [hunk 2]
```diff
@@ -52,12 +52,36 @@ type Model struct {
...
+	Graphite             types.String `tfsdk:"graphite"`
+	MaxDiskThreshold     types.Int64  `tfsdk:"max_disk_threshold"`
+	MetricsFrequency     types.Int64  `tfsdk:"metrics_frequency"`
+	MetricsPrefix        types.String `tfsdk:"metrics_prefix"`
+	MonitoringInstanceId types.String `tfsdk:"monitoring_instance_id"`
+	Plugins              types.List   `tfsdk:"plugins"`
+	Roles                types.List   `tfsdk:"roles"`
+	Syslog               types.List   `tfsdk:"syslog"`
+	TlsCiphers           types.List   `tfsdk:"tls_ciphers"`
+	TlsProtocols         types.String `tfsdk:"tls_protocols"`
```
Comment (joaopalet) [inline stackit/internal/services/rabbitmq/instance/resource.go:67]: It's a bit strange that this is plural, right? But it's the API field name

File: stackit/internal/services/rabbitmq/rabbitmq_acc_test.go [hunk 3]
```diff
@@ -28,50 +28,39 @@ var instanceResource = map[string]string{
...
+	for k, v := range params {
+		if utils.Contains(nonStringParams, k) {
+			parameters += fmt.Sprintf("%s = %s\n", k, v)
+		} else {
+			parameters += fmt.Sprintf("%s = %q\n", k, v)
+		}
+	}
+	parameters += "\n}"
+	return parameters
 }
```
Comment (joaopalet) [inline stackit/internal/services/rabbitmq/rabbitmq_acc_test.go:52]: 👍

File: stackit/internal/services/rabbitmq/instance/resource_test.go [hunk 4]
```diff
@@ -62,7 +62,87 @@ func TestMapFields(t *testing.T) {
...
+					"max_disk_threshold":     types.Int64Null(),
+					"metrics_frequency":      types.Int64Null(),
+					"metrics_prefix":         types.StringNull(),
+					"monitoring_instance_id": types.StringNull(),
+					"plugins":                types.ListNull(types.StringType),
+					"roles":                  types.ListNull(types.StringType),
+					"syslog":                 types.ListNull(types.StringType),
+					"tls_ciphers":            types.ListNull(types.StringType),
+					"tls_protocols":          types.StringNull(),
+				}),
```
Comment (joaopalet) [inline stackit/internal/services/rabbitmq/instance/resource_test.go:78]: I would write these in a `default_params_values` test case, and then the params with the values from the test below could be moved here (and `simple_values_params` could be deleted)

Comment (joaopalet) [reply stackit/internal/services/rabbitmq/instance/resource_test.go:78]: Same for tests below

File: stackit/internal/services/rabbitmq/instance/resource.go (hunk 2 above)
Comment (DiogoFerrao) [reply stackit/internal/services/rabbitmq/instance/resource.go:67]: Very weird 🤷

File: stackit/internal/services/rabbitmq/instance/resource_test.go (hunk 4 above)
Comment (DiogoFerrao) [reply stackit/internal/services/rabbitmq/instance/resource_test.go:78]: Good point, I agree

**Review by joaopalet — APPROVED** [review]

**Pull request merged**